grakn-graql-shell/src/main/java/ai/grakn/graql/JsonSession.java

```diff
@@ -52,7 +52,11 @@
 
     private final Session session;
 
-    private final ExecutorService executor = Executors.newSingleThreadExecutor();
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(runnable -> {
+        Thread thread = Executors.defaultThreadFactory().newThread(runnable);
+        thread.setDaemon(true);
+        return thread;
+    });
```

Comment:
@aelred is there no way to get all thread alive in a given context? We could check what threads are running before the test terminates and make sure that is what is expected.